animation_nodes/nodes/generic/mix_data.py

```diff
@@ -25,27 +43,73 @@ class MixDataNode(bpy.types.Node, AnimationNode):
+                yield "f = min(max(factor, 0.0), 1.0)"
+            else:
+                yield "f = factor"
+            yield getMixCode(self.dataType, "a", "b", "f", "result")
+
+    def execute_MixDataList(self, mix1, mix2, factor):
+        if self.clampFactor:
+            if self.useFactorList:
+                factor.clamp(0, 1)
+            else:
+                factor = min(max(factor, 0.0), 1.0)
+
+        factors = VirtualDoubleList.create(factor, 0)
+
+        if self.dataType == "Float":
+            mix1s, mix2s = VirtualDoubleList.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(mix1s, mix2s, factors)
+            return mixDoubleLists(mix1s, mix2s, factors, amount)
+        elif self.dataType == "Vector":
+            mix1s, mix2s = VirtualVector3DList.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(mix1s, mix2s, factors)
+            return mixVectorLists(mix1s, mix2s, factors, amount)
+        elif self.dataType == "Quaternion":
+            mix1s, mix2s = VirtualQuaternionList.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(factors)
+            return mixQuaternionLists(mix1s, mix2s, factors, amount)
+        elif self.dataType == "Matrix":
+            mix1s, mix2s = VirtualMatrix4x4List.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(mix1s, mix2s, factors)
+            return mixMatrixLists(mix1s, mix2s, factors, amount)
+        elif self.dataType == "Color":
+            mix1s, mix2s = VirtualColorList.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(mix1s, mix2s, factors)
+            return mixColorLists(mix1s, mix2s, factors, amount)
+        elif self.dataType == "Euler":
+            mix1s, mix2s = VirtualEulerList.createMultiple((mix1, 0), (mix2, 0))
+            amount = VirtualDoubleList.getMaxRealLength(mix1s, mix2s, factors)
+            return mixEulerLists(mix1s, mix2s, factors, amount)
 
 def getMixCode(dataType, mix1 = "a", mix2 = "b", factor = "f", result = "result"):
-    if dataType in ("Float", "Vector", "Quaternion"): return "{} = {} * (1 - {}) + {} * {}".format(result, mix1, factor, mix2, factor)
-    if dataType == "Matrix": return "{} = {}.lerp({}, {})".format(result, mix1, mix2, factor)
-    if dataType == "Color": return "{} = [v1 * (1 - {}) + v2 * {} for v1, v2 in zip({}, {})]".format(result, factor, factor, mix1, mix2)
-    if dataType == "Euler": return "{} = animation_nodes.utils.math.mixEulers({}, {}, {})".format(result, mix1, mix2, factor)
+    if dataType in ("Float", "Vector", "Quaternion"): return f"{result} = {mix1} * (1 - {factor}) + {mix2} * {factor}"
+    if dataType == "Matrix": return f"{result} = {mix1}.lerp({mix2}, {factor})"
+    if dataType == "Color": return f"{result} = [v1 * (1 - {factor}) + v2 * {factor} for v1, v2 in zip({mix1}, {mix2})]"
```

Comment:
The return type should be of type Color here.

Reply:
Done.